bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/AddonSuggestionService.java

```diff
@@ -112,49 +94,56 @@ protected void removeAddonFinderService(AddonFinderService addonFinderService) {
 
     @Modified
     public void modified(@Nullable final Map<String, Object> config) {
-        baseFinderConfig.forEach((finder, currentEnabled) -> {
-            String cfgParam = SUGGESTION_FINDER_CONFIGS.get(finder);
-            if (cfgParam != null) {
-                boolean newEnabled = (config != null)
-                        ? ConfigParser.valueAsOrElse(config.get(cfgParam), Boolean.class, true)
-                        : currentEnabled;
-                if (currentEnabled != newEnabled) {
-                    String type = SUGGESTION_FINDER_TYPES.get(finder);
-                    AddonFinderService finderService = addonFinderService;
-                    if (type != null && finderService != null) {
-                        logger.debug("baseFinderConfig {} {} = {} => updating from {} to {}", finder, cfgParam,
-                                config == null ? "null config" : config.get(cfgParam), currentEnabled, newEnabled);
-                        baseFinderConfig.put(finder, newEnabled);
-                        if (newEnabled) {
-                            finderService.install(type);
+        if (config != null) {
+            AddonFinderService finderService = addonFinderService;
+            baseFinderConfig.forEach((finder, currentEnabled) -> {
+                String cfgParam = SUGGESTION_FINDER_CONFIGS.get(finder);
+                if (cfgParam != null) {
+                    boolean newEnabled = ConfigParser.valueAsOrElse(config.get(cfgParam), Boolean.class, true);
+                    if (currentEnabled != newEnabled) {
+                        String type = SUGGESTION_FINDER_TYPES.get(finder);
+                        if (type != null) {
+                            logger.debug("baseFinderConfig {} {} = {} => updating from {} to {}", finder, cfgParam,
+                                    config.get(cfgParam), currentEnabled, newEnabled);
+                            baseFinderConfig.put(finder, newEnabled);
```

Comment:
Before your change, we updated baseFinderConfig only when finderService was not null. Now you do it even when finderService is null. I am not sure it is correct because it could lead to finderService.install being nevcer called ?